### PR TITLE
Ability to add extra user data for lucene commits

### PR DESCRIPTION
**Pull request by kingherc**
Relates ES-5058

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java [hunk 1]
```diff
@@ -202,7 +202,7 @@ static void addIndices(
              * document-level semantics.
              */
             writer.setLiveCommitData(() -> {
-                final Map<String, String> liveCommitData = Maps.newMapWithExpectedSize(3);
+                final Map<String, String> liveCommitData = Maps.newMapWithExpectedSize(4);
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java:205]: :+1:

**Review by idegtiarenko — APPROVED** [review]

**Comment by kingherc** [comment]
Thanks @idegtiarenko ! I would also like to ask @Tim-Brooks 's approval before merging.

**Review by kingherc — COMMENTED** [review]
So based on discussion, I reverted most changes and this now extends only the InternalEngine for custom extra commit user data. Feel free to review again @Tim-Brooks , @idegtiarenko !

**Review by idegtiarenko — APPROVED** [review]

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Pull request merged**